tests/manage/z_cluster/test_rook_operator_restart_during_mon_failover.py

```diff
@@ -0,0 +1,127 @@
+from ocs_ci.framework import config
+from ocs_ci.ocs import constants
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    tier4a,
+    bugzilla,
+    skipif_ocs_version,
+    skipif_external_mode,
+    ignore_leftovers,
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier4a
+@skipif_external_mode
+@skipif_ocs_version("<4.6")
+@bugzilla("1959983")
+@ignore_leftovers
+@pytest.mark.polarion_id("OCS-2572")
+class TestDrainNodeMon(ManageTest):
+    """
+    1.Get worker node name where monitoring pod run
+    2.Verify pdb status, disruptions_allowed=1, max_unavailable_mon=1
+    3.Drain node where monitoring pod run
+    4.Verify pdb status, disruptions_allowed=0, max_unavailable_mon=1
+    5.Verify the number of mon pods is 3 for (1400 seconds)
+    6.Respin  rook-ceph operator pod
+    7.Uncordon the node
+    8.Wait for mon and osd pods to be on running state
+    9.Verify pdb status, disruptions_allowed=1, max_unavailable_mon=1
+
+    """
+
+    def test_rook_operator_restart_during_mon_failover(self, node_drain_teardown):
+        """
+        Verify the number of monitoring pod is three when drain node
+
+        """
+        sample = TimeoutSampler(
+            timeout=100,
+            sleep=10,
+            func=verify_pdb_mon,
+            disruptions_allowed=1,
+            max_unavailable_mon=1,
+        )
+        if sample.wait_for_func_status(result=True):
+            assert "the expected pdb state is not equal to actual pdb state"
+
+        log.info("Get worker node name where monitoring pod run")
+        mon_pod_objs = get_mon_pods()
+        node_name = mon_pod_objs[0].data["spec"]["nodeName"]
+
+        drain_nodes([node_name])
+
+        sample = TimeoutSampler(
+            timeout=100,
+            sleep=10,
+            func=verify_pdb_mon,
+            disruptions_allowed=0,
+            max_unavailable_mon=1,
+        )
+        if sample.wait_for_func_status(result=True):
+            assert "the expected pdb state is not equal to actual pdb state"
+
+        log.info("Verify the number of mon pods is 3")
+        sample = TimeoutSampler(timeout=1400, sleep=10, func=self.check_mon_pods_eq_3)
+        if sample.wait_for_func_status(result=True):
+            assert "There are more than 3 mon pods."
+
+        log.info("Respin pod rook-ceph operator pod")
+        rook_ceph_operator_pod_obj = get_operator_pods()
+        rook_ceph_operator_pod_obj[0].delete()
+
+        schedule_nodes([node_name])
+
+        logging.info("Wait for mon and osd pods to be on running state")
+        pod = OCP(kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"])
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.MON_APP_LABEL,
+            resource_count=3,
+            timeout=100,
+        )
+        assert pod.wait_for_resource(
+            condition="Running",
+            selector=constants.OSD_APP_LABEL,
+            resource_count=3,
```

Comment:
There are cases when the number of the osd pods can be more than 3. 
Maybe we can change it to `resource_count=len(get_osd_pods())` and then we need to add an import of the function `get_osd_pods`. If we expect that the osd pods will be 3 when the test runs, there is no need to change this line.

Reply:
+1 to Itzhak's suggestion. Let's not assume that there will always be 3 osds.

Reply:
@yitzhak12 
As we discussed, I used "wait_for_pods_to_be_running" function